### PR TITLE
feat: support multiple version module as activated

### PR DESCRIPTION
**Pull request by gaosaroma**
增加 conf/ark/bootstrap.properties  中的配置：`sofa.ark.activate.multi.biz.version.enable"`：
- false(默认值)
- true
当 sofa.ark.activate.multi.biz.version.enable = true 时，安装新模块不会修改旧模块的状态，以此允许模块的多个版本同时以 ACTIVATED 状态提供服务。

## Summary by CodeRabbit

- **New Features**
  - Enhanced control flow for activating business modules, allowing for the retention of the old module state.
  - Introduced a new constant to manage the activation of multiple versions of business modules.

- **Bug Fixes**
  - Improved error handling during startup failures for better state management.

- **Tests**
  - Expanded test coverage for the installation of business modules, ensuring previous versions remain activated when applicable.
  - Added new test cases for handling multiple versions of the same business module, including verification of module states.

### CODE REVIEW

File: sofa-ark-parent/core/spi/src/main/java/com/alipay/sofa/ark/spi/constant/Constants.java [hunk 1]
```diff
@@ -208,6 +208,12 @@ public class Constants {
      * uninstall the biz if it starts up failed
      */
     public final static String       AUTO_UNINSTALL_WHEN_FAILED_ENABLE             = "sofa.ark.auto.uninstall.when.failed.enable";
+
+    /**
+     *
+     */
+    public final static String       KEEP_OLD_MODULE_STATE                         = "sofa.ark.keep.old.module.state";
```
Comment (lvjing2) [inline sofa-ark-parent/core/spi/src/main/java/com/alipay/sofa/ark/spi/constant/Constants.java:215]: sofa.ark.biz.multi.version.activated?

Comment (gaosaroma) [reply sofa-ark-parent/core/spi/src/main/java/com/alipay/sofa/ark/spi/constant/Constants.java:215]: 可以

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/model/BizModel.java [hunk 2]
```diff
@@ -351,7 +352,17 @@ private void doStart(String[] args, Map<String, String> envs) throws Throwable {
...
+        if (activateMultiBizVersion) {
+            setBizState(BizState.ACTIVATED, StateChangeReason.STARTED,
+                String.format("started a new biz: %s", this.getIdentity()));
+            return;
+        }
+
         if (Boolean.getBoolean(Constants.ACTIVATE_NEW_MODULE)) {
+            // active latest
             Biz currentActiveBiz = bizManagerService.getActiveBiz(bizName);
             if (currentActiveBiz == null) {
```
Comment (lvjing2) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/model/BizModel.java:367]: 将 currentActiveBiz 的逻辑与当前 biz 的逻辑分离，代码结构会清晰些

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**